qiskit/algorithms/evolvers/evolution_problem.py

```diff
@@ -41,7 +41,9 @@ def __init__(
         Args:
             hamiltonian: The Hamiltonian under which to evolve the system.
             time: Total time of evolution.
-            initial_state: Quantum state to be evolved.
+            initial_state: The quantum state to be evolved for methods like Trotterization.
+                For variational time evolutions, where the evolution happens in an ansatz,
+                this argument is not required.
```

Comment:
True, I'll add that 👍🏻

Reply:
Actually -- that's not possible right now since not all initial states can be adjoint. That's because opflow uses `Intialize` to convert a statevector into a circuit operation. We can fix that by making `VectorStateFn` use the new `StatePreparation` instruction and then we can also allow initial states in p-VQD. I'll open an issue to fix the opflow thing and then update p-VQD, but maybe that's better kept as a follow-up 🙂